src/core/handle.ml

```diff
@@ -133,6 +134,89 @@ let handle_require_as : popt -> sig_state -> Path.t -> ident -> sig_state =
   let path_map = PathMap.add p id.elt ss.path_map in
   {ss with aliases; path_map}
 
+(** Symbol properties needed for the signature are packed together *)
+type sig_symbol =
+  { expo   : expo        ;
+    prop   : prop        ;
+    mstrat : match_strat ;
+    x      : ident       ;
+    a      : term        ;
+    impl   : bool list   ;
+    d      : term option ; }
+
+(** Wrapper around Sig_state.add_symbol using the sig_symbol type *)
+let add_symbol : sig_state -> sig_symbol -> sig_state =
```

Comment:
Why not moving the type sig_symbol to sig_state.ml and redefine Sig_state.add_symbol so that it takes the type `: sig_state -> sig_symbol -> sig_state`?

Reply:
fixed in dfb478e